crates/task-impls/src/da.rs

```diff
@@ -394,18 +394,18 @@ where
 
                 return None;
             }
-            HotShotEvent::BlockReady(block, view) => {
+            HotShotEvent::BlockReady(payload, metadata, view) => {
                 self.committee_exchange
                     .network()
                     .inject_consensus_info(ConsensusIntentEvent::CancelPollForTransactions(*view))
                     .await;
 
-                let payload_commitment = block.commit();
+                let payload_commitment = payload.commit();
                 let signature = self
                     .committee_exchange
                     .sign_da_proposal(&payload_commitment);
                 let data: DAProposal<TYPES> = DAProposal {
-                    block_payload: block.clone(),
+                    block_payload: payload.clone(),
```

Comment:
Should we send ` payload.encode()` here? As we discussed before the DA member task will need to validate the raw payload anyways, so we can save them a step of calling `encode()` to get the raw payload, and also avoid serializing extra metadata in the structured form of `payload` that the receiver of this message cannot trust anyways.

Reply:
Speaking of which, elsewhere in this file when we handle `DAProposalRecv`, we need to call `decode()` to compute the commitment from the raw payload, rather than trusting the commitment in the proposal as we do now

Reply:
Yeah, I'm going to do this in the next PR. Issue: https://github.com/EspressoSystems/HotShot/issues/2026.